proto/extensions.proto

```diff
@@ -32,3 +32,28 @@ message DownloadIntentExtension {
   // download everything.
   bool everything = 2;
 }
+
+message MapShareExtension {
+  // The URL of the server to perform map shares over
+  string url = 1;
+  // The ID of the device that sent the map share 
+  string senderDeviceId = 2;
+  // The name of the device that sent the map share 
+  string senderDeviceName = 3;
+  // The ID of the map share
+  string shareId = 4;
+  // The name of the map being shared
+  string mapName = 5;
+  // The ID of the map being shared 
+  string mapId = 6;
+  // The timestamp when the map share invite was received
+  uint64 receivedAt = 7;
+  // The bounding box of the map data being shared 
+  repeated double bounds = 8;
+  // The minimum zoom level of the map data being shared 
+  int32 minzoom = 9;
+  // The maximum zoom level of the map data being shared
+  int32 maxzoom = 10;
```

Comment:
These will need an additional validation step - bounds must be a tuple of length 4 within these max: `[-180, -85.051129, 180, 85.051129]` (that's the max for spherical mercator projection)

Zoom must be between 0-22.